core/src/storage/impls/state_manager.rs

```diff
@@ -356,6 +356,19 @@ impl StateManagerTrait for StateManager {
     }
 }
 
+// FIXME This is a quick fix for the ref loop of storage_manager. We should
+// eliminate this loop. Arc<StorageManager> -> snapshot_associated_mpts_by_epoch
+// contains -> Arc<DeltaMpt> -> delta_mpts_releaser -> storage_manager ->
+// Arc<StorageManager>
+impl Drop for StateManager {
+    fn drop(&mut self) {
```

Comment:
Thank you for discovering this cycle.
Make it weak in DeltaDbReleaser should be the real fix.

I don't understand how this change fixed the issue, since the ref count never drop to 0.
There is a drop method for StorageManager which does something useful, so let me a push real fix quickly.

Reply:
I need some time to figure out, if Weak pointer upgrade works when the Arc itself is being dropped.

Reply:
It looks like that in rust weak pointer upgrade doesn't work when Arc itself is being dropped. WTH..